call-home/src/bin/callhome/collector/report_models.rs

```diff
@@ -184,3 +215,93 @@ fn get_pools_size_vector(pools: Vec<openapi::models::Pool>) -> Vec<u64> {
     }
     pools_size_vector
 }
+
+/// Record of events populated from prometheus.
+#[derive(Debug, Default, Clone)]
+/// Stats fetched from prometheus data.
+pub struct EventsRecord {
+    pub record: Vec<Sample>,
+}
+
+impl EventsRecord {
+    pub fn new(data: Vec<Sample>) -> Self {
+        Self { record: data }
+    }
+
+    /// Maps the prometheus data to EventData
+    pub fn event_record(&self) -> errors::Result<EventData> {
+        let mut event_record = EventData::default();
+        let mut iterator = self.record.iter();
+        if let Some(rec) = iterator.next() {
+            let val = rec.labels.get("action").unwrap();
```

Comment:
` pub fn event_record(&self) -> errors::Result<EventData>` actually populates `EventData`. which is used every where . Having separate function to get the required values will repeat the code. The better implementation is done in my code , wherein the rec is parsed and EventData is populated once from it.

Reply:
But you already have separate functions here to get action and component.
Having the new_type wrapper won't add a separate function, instead it will combine the impl of these in a neat place.
https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Reply:
done